model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -1289,19 +1280,20 @@ def run_predictor_step(
             #  - $\exnerprimedz{\ntilde}{\c}{\k}$ : z_dexner_dz_c_1
             #  - $\Whor$ : c_lin_e
             #
-            self._compute_horizontal_gradient_of_exner_pressure_for_nonflat_coordinates(
-                inv_dual_edge_length=self._edge_geometry.inverse_dual_edge_lengths,
-                z_exner_ex_pr=self.z_exner_ex_pr,
-                ddxn_z_full=self._metric_state_nonhydro.ddxn_z_full,
-                c_lin_e=self._interpolation_state.c_lin_e,
-                z_dexner_dz_c_1=self.z_dexner_dz_c_1,
-                z_gradh_exner=z_fields.z_gradh_exner,
-                horizontal_start=self._start_edge_nudging_level_2,
-                horizontal_end=self._end_edge_local,
-                vertical_start=self._vertical_params.nflatlev,
-                vertical_end=gtx.int32(self._vertical_params.nflat_gradp + 1),
-                offset_provider=self._grid.offset_providers,
-            )
+            if self._vertical_params.nflatlev < gtx.int32(self._vertical_params.nflat_gradp + 1):
```

Comment:
Can it happen that `nflatlev>= nflat_gradp + 1` ? Shouldn't the condition rather be checked when computing these values at initialization?  The other questions is does gt4py crash for a "negative" or "empty" domain?

Reply:
Yes, it crashed for an empty domain. We should probably fix that on the GridTools side.

Reply:
Yes, I would suggest so. Anyway, we can leave this as a workaround for now. Can you open an issue in GT4Py?

Reply:
On a second thought, I am not sure if it should be legal, because that could also hide user errors. At least GT4Py should warn that it doesn't execute a program. Here is the issue https://github.com/GridTools/gt4py/issues/1891

Reply:
Thanks.  I agree it could hide errors, but there are valid cases for it: for example if you run something like APE every layer is flat, so if you try to compute something from some layer up to the first flat layer, that might collapse (what probably happens in the example here) or in the horizontal if you compute something only on the boundaries, but you have a global model, then there are no boundaries. The user would then have to always do an extra
`if domain-is-not-empty` check what the if condition above essentially is.